scripts/uaa_lock_user.rb

```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/env ruby
+
+require 'optparse'
+require 'uaa'
+require 'ostruct'
+
+STATE_MAP = {
+  true => "unlocked",
+  false => "locked",
+}.freeze
+
+def parse_args
+  desired_active_state = false
+  settings = OpenStruct.new(
+    target: "",
+    token: "",
+    insecure: false,
+  )
+
+  optparse = OptionParser.new do |opts|
+    opts.banner = <<EOS
+Usage: uaa_lock_user.rb [options] USERNAME
+
+You need to provide the following environment variables:
+
+  TARGET=$(cf curl /v2/info | jq -r .token_endpoint)
+  TOKEN=$(cf oauth-token)
+
+EOS
+
+    opts.on("--skip-ssl-validation", "Skip verification of the API endpoint. Not recommended!") do
+      settings.insecure = true
+    end
+
+    opts.on("-u", "--unlock", "Unlock user") do
+      desired_active_state = true
+    end
+  end
+  optparse.parse!
+
+  if ARGV.size != 1 || !ENV.include?("TARGET") || !ENV.include?("TOKEN")
+    puts optparse
+    exit 2
+  end
+
+  username = ARGV[0]
+  settings.target = ENV.fetch("TARGET")
+  settings.token = ENV.fetch("TOKEN")
+
+  [username, desired_active_state, settings]
+end
+
+def update_user(username, active, uaac)
+  query = { filter: "username eq '#{username}'" }
+  users = uaac.all_pages(:user, query)
+
+  if users.empty?
+    abort "Username not found"
+  elsif users.size > 1
+    abort "Username is not unique"
+  end
+  user = users.first
+
+  if user.fetch('active') != active
+    user['active'] = active
+    uaac.patch(:user, user)
+  end
+
+  "#{STATE_MAP.fetch(active)} #{username}"
```

Comment:
Not that I think its worth changing now, but i'd be tempted to return 2 values here rather than a string, as it feels more flexible